### PR TITLE
Some suggestions in the Scala 3 book (2)

### PR DESCRIPTION
**Pull request by julienrf**
I’ve reviewed the following chapters:
- a first look at types
- control structures
- domain modeling

Here are some simple suggestions, or typos fixes.

Maybe @alvinj you are interested in reviewing?

### CODE REVIEW

File: _overviews/scala3-book/control-structures.md [hunk 1]
```diff
@@ -144,26 +144,14 @@ scala> for i <- ints do println(i)
...
-// option 2
-for (i <- ints)
-  val x = i * 2
-  println(s"i = $i, x = $x")
-
-// option 3
-for (i <- ints) {
-  val x = i * 2
-  println(s"i = $i, x = $x")
-}
```
Comment (julienrf) [inline _overviews/scala3-book/control-structures.md:166]: I suggest not showing too many options, unless there is a clear rule that says which one to pick when. Here, I would just show the preferred option.

File: _overviews/scala3-book/domain-modeling-tools.md [hunk 2]
```diff
@@ -33,27 +33,27 @@ class Movie(var name: String, var director: String, var year: Int)
...
-    this()
+  public Person(String name, String vocation) {
     // assigning the fields
-    name = _name
-    vocation = _vocation
+    this.name = name;
+    this.vocation = vocation;
+  }
+}
 ```
```
Comment (julienrf) [inline _overviews/scala3-book/domain-modeling-tools.md:51]: The original Scala code was not valid. My commit uses a Java version, which makes sense only to Java users, so I’m not sure it’s a good idea, maybe we should just remove that code snippet?

**Comment by alvinj** [comment]
> Maybe @alvinj you are interested in reviewing?

Hi @julienrf. Yes, I’ll be glad to review these, though it might be Saturday before I can get to them.

**Comment by alvinj** [comment]
> though it might be Saturday before I can get to them.

Sorry, please make that Sunday, June 13th. (I’ll review them then.)

File: _overviews/scala3-book/control-structures.md (hunk 1 above)
Comment (alvinj) [reply _overviews/scala3-book/control-structures.md:166]: Sounds good to me!

File: _overviews/scala3-book/domain-modeling-oop.md [hunk 3]
```diff
@@ -89,12 +89,11 @@ When designing software in Scala, it’s often helpful to only consider using cl
 NOTE: I think “leaves” may technically be the correct word to use, but I prefer “leafs.”
 {% endcomment %}
 
-```text
-traits               T1   T2           ...  T3
-composed traits      S extends T1, T2  ...  S extends T2, T3
-classes              C extends S, T3
-instances            new C
-```
+| Traits          | `T1`, `T2`, `T3`
```
Comment (alvinj) [inline _overviews/scala3-book/domain-modeling-oop.md:92]: At the moment I can’t run Jekyll on my computer to render this, but if that table syntax works, this is fine by me.

File: _overviews/scala3-book/domain-modeling-tools.md (hunk 2 above)
Comment (alvinj) [reply _overviews/scala3-book/domain-modeling-tools.md:51]: I’m sorry about that initial Scala code, that definitely wasn’t the right approach.

I can go either way on this. On the one hand, just showing the Scala code shows how concise it is. People who use any other language can make the “conciseness” comparison in their head.

My original logic was (a) I assume that most developers have worked with Java, and (b) I wanted to appeal to Java developers. But again, they can make that comparison in their head. (And if we show Java, why not show Python as well?)

So yes, after thinking about it, I would delete that comparison to Java.

**Review by alvinj — CHANGES_REQUESTED** [review]
All of these changes look great — thanks!

File: _overviews/scala3-book/domain-modeling-oop.md (hunk 3 above)
Comment (julienrf) [reply _overviews/scala3-book/domain-modeling-oop.md:92]: Sorry, I should have included a screenshot. It looks like this:

Screenshot from 2021-06-14 10-27-37

File: _overviews/scala3-book/domain-modeling-tools.md (hunk 2 above)
Comment (julienrf) [reply _overviews/scala3-book/domain-modeling-tools.md]: OK, thanks for sharing your thoughts. I have removed the Java code snippet.

File: _overviews/scala3-book/domain-modeling-oop.md (hunk 3 above)
Comment (alvinj) [reply _overviews/scala3-book/domain-modeling-oop.md:92]: No problem on this. I just had a problem with Jekyll and its tools, and I didn’t have time to look into how to fix it.

**Pull request merged**